### PR TITLE
smart: try to harmonize all smart definitions implementations

### PR DESCRIPTION
**Pull request by hsablonniere**

### CODE REVIEW

**Review by florian-sanders-cc — APPROVED** [review]

File: src/pricing/cc-pricing-product-consumption.smart.js [hunk 1]
```diff
@@ -56,12 +56,13 @@ defineComponent({
...
-      product_lp.value$.subscribe((product) => {
-        component.sections = product.sections;
-      }),
+      product_lp.value$.subscribe((product) => (component.sections = product.sections)),
 
       context$.subscribe(({ productId, zoneId, currency }) => {
 
+        // TODO, we may need to refine this reset
+        component.sections = null;
+
```
Comment (pdesoyres-cc) [inline src/pricing/cc-pricing-product-consumption.smart.js:65]: At some point, we need to reset the `error` property too
`component.error = false;`

Should we do that here or at line 59?

File: src/pricing/cc-pricing-product.smart-addon.js [hunk 2]
```diff
@@ -33,9 +33,17 @@ defineComponent({
       }),
 
       context$.subscribe(({ productId, zoneId, currency, addonFeatures }) => {
+
+        component.name = null;
+        component.icon = null;
+        component.description = null;
+        component.plans = null;
+        component.features = null;
+
```
Comment (pdesoyres-cc) [inline src/pricing/cc-pricing-product.smart-addon.js:42]: At some point, we need to reset the `error` property too
`component.error = false;`

Should we do that here or at line 28?

File: src/pricing/cc-pricing-product.smart-runtime.js [hunk 3]
```diff
@@ -32,9 +32,17 @@ defineComponent({
       }),
 
       context$.subscribe(({ productId, zoneId, currency }) => {
+
+        component.name = null;
+        component.icon = null;
+        component.description = null;
+        component.plans = null;
+        component.features = null;
+
```
Comment (pdesoyres-cc) [inline src/pricing/cc-pricing-product.smart-runtime.js:41]: At some point, we need to reset the `error` property too
`component.error = false;`

Should we do that here or at line 27?

File: src/saas/cc-grafana-info.smart.js [hunk 4]
```diff
@@ -11,14 +11,14 @@ import { sendToApi } from '../lib/send-to-api.js';
 import { defineComponent } from '../lib/smart-manager.js';
 import { addErrorType } from '../lib/utils.js';
 
+// TODO, we need to refactor this one to make it more like the others
```
Comment (pdesoyres-cc) [inline src/saas/cc-grafana-info.smart.js:14]: I don't understand what kind of refactoring should be made here.

File: src/tcp-redirections/cc-tcp-redirection-form.smart.js [hunk 5]
```diff
@@ -78,25 +60,25 @@ defineComponent({
         component.redirections = null;
```
Comment (pdesoyres-cc) [inline src/tcp-redirections/cc-tcp-redirection-form.smart.js:61]: At some point, we need to reset the `error` property too
`component.error = null;`

Should we do that here or at line 39?

File: src/addon/cc-jenkins-info.smart.js [hunk 6]
```diff
@@ -15,40 +15,42 @@ defineComponent({
...
 
       context$.subscribe(({ apiConfig, addonId }) => {
 
         component.error = false;
+        component.jenkinsLink = null;
         component.jenkinsManageLink = null;
         component.versions = null;
 
         if (apiConfig != null && addonId != null) {
           addon_lp.push((signal) => fetchAddon({ apiConfig, signal, addonId }));
```
[the remark this answers is not in the document]
Comment (hsablonniere) [reply src/addon/cc-jenkins-info.smart.js:41]: For this one, we have separated `skeleton` states for these parts so we can load them one by one when they arrive.

File: src/pricing/cc-pricing-product.smart-addon.js (hunk 2 above)
Comment (hsablonniere) [reply src/pricing/cc-pricing-product.smart-addon.js:42]: Thanks for the feedback, I added the error reset but in the context subscribe (not the value subscribe).

File: src/pricing/cc-pricing-product.smart-runtime.js (hunk 3 above)
Comment (hsablonniere) [reply src/pricing/cc-pricing-product.smart-runtime.js:41]: Thanks for the feedback, I added the error reset but in the context subscribe (not the value subscribe).

File: src/tcp-redirections/cc-tcp-redirection-form.smart.js (hunk 5 above)
Comment (hsablonniere) [reply src/tcp-redirections/cc-tcp-redirection-form.smart.js:61]: I don't understand this remark for this example as like 59 already has a reset for the error property.

File: src/invoices/cc-invoice.smart.js [hunk 7]
```diff
@@ -14,29 +14,34 @@ defineComponent({
...
+      invoice_lp.error$.subscribe(() => (component.error = true)),
+      invoice_lp.value$.subscribe((invoice) => (component.invoice = invoice)),
 
       context$.subscribe(({ apiConfig, ownerId, invoiceNumber }) => {
-        if (apiConfig != null && ownerId != null && invoiceNumber != null) {
 
-          component.number = invoiceNumber;
-          component.invoice = null;
+        component.number = invoiceNumber;
+        component.invoice = null;
```
[the remark this answers is not in the document]
Comment (hsablonniere) [reply src/invoices/cc-invoice.smart.js:28]: Thanks for the feedback, I added the error reset but in the context subscribe (not the value subscribe).

**Comment by hsablonniere** [comment]
@pdesoyres-cc I replied to all your feedbacks and pushed an improved version, thanks again :wink:

**Review by pdesoyres-cc — APPROVED** [review]

File: src/tcp-redirections/cc-tcp-redirection-form.smart.js (hunk 5 above)
Comment (pdesoyres-cc) [reply src/tcp-redirections/cc-tcp-redirection-form.smart.js:61]: Sorry for this one

**Comment by pdesoyres-cc** [comment]
Looks goot to me :+1:

**Pull request merged**